### PR TITLE
[WIP]: Make README more interesting

### PR DESCRIPTION
**Pull request by msaroufim**
I want to show people how they can just instrument the training loop, how it works with popular libraries, what the codegen'd triton kernels looks lik. Draft since I haven't tested that the code snippet run or make things faster yet.

Plz ignore this until I mark as ready for review

EDIT: Need to mention cuda graphs more explicitly as one other key optimization

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,9 +18,47 @@ This repository still contains:
...
+TorchDynamo supports arbitrary PyTorch code, control flow, mutation and dynamic shapes. 
 
-You can follow our nightly benchmarks [here](https://github.com/pytorch/torchdynamo/issues/681)
+So this works
+
+```python
+import torch
+import torch._dynamo as dynamo
+model = torch.hub.load('pytorch/vision:v0.10.0', 'resnet50', pretrained=True)
+dynamo.optimize("inductor")(model) # Leverage the inductor training compiler
```
Comment (jansel) [inline README.md:29]: Need to do something with the result?  Perhaps:
`model = ...`

File: README.md [hunk 2]
```diff
@@ -102,21 +140,62 @@ If you do not have the PyTorch source locally, you can alternatively copy the sc
...
     a = torch.cos(x)
     b = torch.sin(y)
     return a + b
 ```
 
-It's also easy to define your own compiler backends in pure python [custom backend](./documentation/custom-backend.md)
+Or as a function
+
+```py
+dynamo_optimize("inductor")(fn)
```
Comment (jansel) [inline README.md:160]: Same

File: README.md [hunk 3]
```diff
@@ -102,21 +140,62 @@ If you do not have the PyTorch source locally, you can alternatively copy the sc
...
+
+# And in longer form you can the above as
+inductor_compiler = dynamo.optimize("inductor")
+inductor_compiler(fn)
```
Comment (jansel) [inline README.md:164]: Same

File: README.md [hunk 4]
```diff
@@ -102,21 +140,62 @@ If you do not have the PyTorch source locally, you can alternatively copy the sc
...
+        self.lin1 = nn.Linear(10,10)
+        self.lin2 = nn.Linear(10,10)
+    def forward(self, x):
+        x = self.lin1(x)
+        x = self.lin2(x)
+        return x
+
+
+mod = ToyModel()
+dynamo.optimize("inductor")(mod)
```
Comment (jansel) [inline README.md:182]: Same

**Comment by msaroufim** [comment]
I will focus on the module API for this README since it seems like the AOT input mutation stuff will land later

**Pull request closed without merging**